src/indexer/merger.rs

```diff
@@ -183,6 +183,28 @@ impl IndexMerger {
                 readers.push(reader);
             }
         }
+        if let Some(sort_by_field) = index_settings.sort_by_field.as_ref() {
```

Comment:
No need to fix: 

Just a remark.
`sort_by_cached_key` would be have been cleaner, but does not allow for error handling.
I almost miss Exceptions.

https://doc.rust-lang.org/std/primitive.slice.html#method.sort_by_cached_key

Reply:
I feel like this is a missing building block in Rust, to convert any function into one which returns a Result. But it would probably add too much complexity at this stage into the language.

Reply:
I think this is a contrarian opinion but I actually like that Java semantics of Exception.